activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -287,7 +288,11 @@ def each_current_configuration(environment)
 
         configurations = ActiveRecord::Base.configurations.values_at(*environments)
         configurations.compact.each do |configuration|
-          yield configuration unless configuration['database'].blank?
+          if configuration['database'].blank?
+            raise DatabaseNotSpecified, 'You tried to run a database rake task, but no database is specified.'
+          else
+            yield configuration
+          end
```

Comment:
I don't think you need an `else` here, as if the first one raise, the stack will be poped